pkg/jx/cmd/util/factory.go

```diff
@@ -70,7 +71,7 @@ func (f *factory) SetBatch(batch bool) {
 
 // CreateJenkinsClient creates a new jenkins client
 func (f *factory) CreateJenkinsClient() (*gojenkins.Jenkins, error) {
-	url := os.Getenv("JENKINS_URL")
```

Comment:
This causes the jenkins auth config file to use the internal jenkins service URL when run in cluster.